flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1270,6 +1272,42 @@ public void run() {
 	}
 
 	/**
+	 * Discards the given pending checkpoint because of the given cause.
+	 *
+	 * @param pendingCheckpoint to discard
+	 * @param cause for discarding the checkpoint
+	 */
+	private void discardCheckpoint(PendingCheckpoint pendingCheckpoint, @Nullable Throwable cause) {
+		Thread.holdsLock(lock);
```

Comment:
Should that be an `assert(Thread.holdsLock(lock));` or a `Preconditions.checkState(Thread.holdsLock(lock));`?

Reply:
Yes definitely.